paddle/fluid/operators/repeat_interleave_op.h

```diff
@@ -0,0 +1,196 @@
+      std::vector<int> index_vec(index_size);
+      for (int i = 0; i < inputs.dims()[dim]; i++) {
+        std::fill_n(index_vec.begin() + i * repeats, repeats, i);
+      }
+      index.Resize(framework::make_ddim({index_size}));
+      paddle::framework::TensorFromVector<int>(index_vec, &index);
+
+      auto output_dim = framework::vectorize(inputs.dims());
+      output_dim[dim] = index_size;
+      output->Resize(framework::make_ddim(output_dim));
+
+      IndexSelectInner<DeviceContext, T, int>(context, &inputs, index, output,
+                                              dim);
+    } else {
+      PADDLE_THROW(platform::errors::InvalidArgument(
+          "repeats must given with RepeatsTensor (tensor) or repeats (int)"));
+    }
+  }
+};
+
+template <typename DeviceContext, typename T>
+class RepeatInterleaveGradKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& context) const override {
+    auto* x_grad =
+        context.Output<framework::LoDTensor>(framework::GradVarName("X"));
+    auto* out_grad =
+        context.Input<framework::LoDTensor>(framework::GradVarName("Out"));
+
+    int dim = context.Attr<int>("dim");
+    if (dim < 0) {
+      dim += out_grad->dims().size();
+    }
+
+    int repeats = context.Attr<int>("Repeats");
+    framework::LoDTensor index;
+    if (context.HasInput("RepeatsTensor")) {
+      auto repeats_tensor =
+          context.Input<framework::LoDTensor>("RepeatsTensor");
+      const auto& index_type = repeats_tensor->type();
+
+      bool index_type_match = index_type == framework::proto::VarType::INT32 ||
+                              index_type == framework::proto::VarType::INT64;
+      PADDLE_ENFORCE_EQ(
+          index_type_match, true,
+          platform::errors::InvalidArgument(
+              "Input(Repeats) holds the wrong type, it holds %s, but "
+              "desires to be %s or %s",
+              paddle::framework::DataTypeToString(index_type),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT32),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT64)));
+
+      if (index_type == framework::proto::VarType::INT32) {
+        RepeatsTensor2IndexTensor<DeviceContext, int>(*repeats_tensor, &index);
+        IndexSelectGradInner<DeviceContext, T, int>(context, *out_grad, index,
+                                                    x_grad, dim);
+      } else if (index_type == framework::proto::VarType::INT64) {
+        RepeatsTensor2IndexTensor<DeviceContext, int64_t>(*repeats_tensor,
+                                                          &index);
+        IndexSelectGradInner<DeviceContext, T, int64_t>(context, *out_grad,
+                                                        index, x_grad, dim);
+      }
+    } else if (repeats > 0) {
+      int index_size = x_grad->dims()[dim] * repeats;
```

Comment:
x_grad->dims()[dim] will return data type of int64_t, index_size should be define as int64_t ?